contracts/Claims.sol

```diff
@@ -0,0 +1,52 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity ^0.8.20;
+
+import {Currency, CurrencyLibrary} from "./types/Currency.sol";
+import {IClaims} from "./interfaces/IClaims.sol";
+
+/// An intentionally barebones balance mapping only supporting mint/burn/transfer
+contract Claims is IClaims {
+    using CurrencyLibrary for Currency;
+
+    // Mapping from Currency id to account balances
+    mapping(uint256 => mapping(address => uint256)) public balances;
```

Comment:
Definetly a bit redundant I agree, however fwiw I like having all external functions (balanceOf and transfer) have inputs of `Currency` type, and all internal ones (_mint, _burn, etc.) use the uint256 id representation. Like since periphery contracts will be interacting with Currencies anyways, easier to let them fetch balances without having to unwrap?

Reply:
Going to make the mapping private though, pool manager will never directly update the mapping

Reply:
On my first comment above, Mark pointed out that we can just make the mapping key Currency since we don't need to comply with the 1155 style uint256 id anymore. Then I do think `balanceOf` is a little redundant but IMO I still like the pattern of calling balanceOf over accessing balances directly? I checked the bytecode savings and its only 20 b

Reply:
We also might want to end up doing a custom storage setup where we do a hashed concat of the Currency + account

Reply:
yeah i dont mind it so much now that balances is private